docs/operating-scylla/security/authorization.rst

```diff
@@ -91,6 +91,30 @@ If the option is used and the role exists, the statement is a no-op::
     CREATE ROLE other_role;
     CREATE ROLE IF NOT EXISTS other_role;
 
+Creating a role with a hashed password
+``````````````````````````````````````
+
+When you create a role with a password, the password is hashed by ScyllaDB and only then inserted into the database.
+However, there also exists a version of the statement allowing you to create a role with a pre-hashed password; the syntax:
+
+.. code-block::
+
+   create_role_statement: CREATE ROLE [ IF NOT EXISTS ] `role_name`
+                        : WITH HASHED PASSWORD `hashed_password`
+                        :     ( AND `role_option` )*
+   hashed_password: `string`
+   role_option: LOGIN '=' `boolean`
+              :| SUPERUSER '=' `boolean`
+              :| OPTIONS '=' `map_literal`
+
+When that statement is executed, ScyllaDB will not hash the provided hashed password; it will be inserted into
+the database directly, with no additional processing of it. The statement only works with an authenticator that
+uses passwords, e.g. PasswordAuthenticator. It can only be executed by a role with the superuser privilege.
+
+The provided hashed password must be an encrypted string using the library function `crypt(3)` specified by POSIX,
+and it must use one of the encryption algorithms that ScyllaDB supports: bcrypt, sha512crypt, sha256crypt, md5crypt.
```

Comment:
Would it be possible to provide an example on how to produce such a hash?

Reply:
I could try to add a block of C code with comments showcasing how to use `crypt(3)`. It would be way more difficult to explain it in words alone, though.

Reply:
Note: in Scylla, we actually use `crypt_r(3)`, which is an alternative for `crypt(3)`, but as far as I can tell the only difference between the functions is `crypt_r(3)` is re-entrant (for that, it takes an additional parameter that it operates on instead of using a static buffer).